libraries/hotstuff/include/eosio/hotstuff/qc_chain.hpp

```diff
@@ -202,20 +251,23 @@ namespace eosio::hotstuff {
       };
 
       bool _chained_mode = false;
+
       block_id_type _block_exec;
       block_id_type _pending_proposal_block;
+      safety_state _safety_state;
```

Comment:
Need to update all `safety_state` to `afety_states` to get compiled.

Reply:
I'm going to revert the `std::optional` patch. Either we commit this as file persistence ticket of a previous safety state logic, or we roll in reviewing and fixing the safety state logic with the file persistence in this same PR.

Reply:
I think you can commit this as previous safety state logic after CI is passed. Then in a new PR fix the remaining issues. This way we can move forward quickly.

Reply:
Yes, I agree, you can merge this. I think it is clear it is not the final version but this moves us forward.